### PR TITLE
fix: Instructions to build & Update dependencies to work properly

### PR DESCRIPTION
**Pull request by kabylake1**
* [x] I understand and have followed the [contribution guide](https://github.com/revoltchat/revolt/discussions/282)
* [x] I have tested my changes locally and they are working as intended
* [x] These changes do not have any notable side effects on other Revolt projects

As per issue #5.

Not sure if this is what's meant to be shown?

Apparently the server list is just the ominous phrase "Among Us" being repeated over and over in sequence?

Again I haven't touched anything other than the build and documenting how to make it work. So...

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,39 @@
...
+
+Unfortunely, in some cases it is difficult to utilize `meson install` properly. In this case the user can just specify the prefix to the current build directory:
+```sh
+# inside builddir
+meson configure -D`pwd`
+```
+
+When this is done, the resources will be located inside the `share` folder on the build directory.
+
+If not installing the program with `meson install` then it is nescesary to notify glib of the schemas residing inside the `data/` folder; so the client shall be run as:
```
Comment (Rexogamer) [inline README.md]:
```suggestion
If the app is not installed with `meson install`, you'll need to tell glib that the schemas reside inside the `data/` folder; therefore, run the client with this command:
```

File: README.md [hunk 2]
```diff
@@ -2,6 +2,39 @@
...
+* (The Rust toolchain)[https://www.rust-lang.org/tools/install]
+* (The GTK4 library)[https://github.com/ToshioCP/Gtk4-tutorial/blob/main/gfm/sec2.md]
+* (The Adwaita library)[https://github.com/GNOME/libadwaita]
+
+It's then essential to setup the build directory with meson:
+```sh
+meson setup builddir
+```
+
+Once this is succesful; It's then proper to perform the compilation procedure:
```
Comment (Rexogamer) [inline README.md]:
```suggestion
Once this is done, compile the app:
```

File: README.md [hunk 3]
```diff
@@ -2,6 +2,39 @@
...
+## Installation
+
+The following components are required beforehand:
+
+* (The meson build system)[https://mesonbuild.com/Quick-guide.html]
+* (The Rust toolchain)[https://www.rust-lang.org/tools/install]
+* (The GTK4 library)[https://github.com/ToshioCP/Gtk4-tutorial/blob/main/gfm/sec2.md]
+* (The Adwaita library)[https://github.com/GNOME/libadwaita]
+
+It's then essential to setup the build directory with meson:
```
Comment (Rexogamer) [inline README.md]:
```suggestion
After installing the necessary components, set up `builddir`:
```

File: README.md [hunk 4]
```diff
@@ -2,6 +2,39 @@
 
 GTK4 Revolt client
 
+## Installation
+
+The following components are required beforehand:
```
Comment (Rexogamer) [inline README.md]:
```suggestion
First, make sure the following components are installed:
```

File: README.md [hunk 5]
```diff
@@ -2,6 +2,39 @@
...
+
+Once this is succesful; It's then proper to perform the compilation procedure:
+```sh
+cd builddir && meson compile
+```
+
+Unfortunely, in some cases it is difficult to utilize `meson install` properly. In this case the user can just specify the prefix to the current build directory:
+```sh
+# inside builddir
+meson configure -D`pwd`
```
Comment (Rexogamer) [inline README.md]: I assume this needs a space?
```suggestion
meson configure -D `pwd`
```

**Review by Rexogamer — COMMENTED** [review]
Your *suspicions* were correct; those are placeholders. 😉 

Anyway, a few suggestions:

**Review by Rexogamer — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -2,6 +2,39 @@
...
+
+Once this is done, compile the app:
+```sh
+cd builddir && meson compile
+```
+
+Unfortunely, in some cases it is difficult to utilize `meson install` properly. In this case the user can just specify the prefix to the current build directory:
+```sh
+# inside builddir
+meson configure -D `pwd`
```
Comment (eli-schwartz) [inline README.md:27]: I'm not sure what this is supposed to do. The `-D` command-line flag. expects to receive an argument such as `option_name=value`, and you're giving it the current directory which is a value without an option name.

Guessing based on the context of "the `share` folder on the build directory" that this is supposed to be `-D prefix=$(pwd)`?

Comment (eli-schwartz) [reply README.md:27]: It seems to me like you probably don't want to do any of this. Instead I would investigate https://mesonbuild.com/Commands.html#devenv which you can also use in combination with `meson.add_devenv()` to enhance the devenv shell with e.g. XDG_DATA_DIRS, use gnome.compile_schemas(), and so on.

**Comment by Rexogamer** [comment]
(this PR is no longer needed as we are restarting from scratch in JS)

**Pull request closed without merging**